firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -778,13 +782,52 @@ public void saveNamedQuery(NamedQuery namedQuery, ImmutableSortedSet<DocumentKey
         "Get named query", () -> bundleCache.getNamedQuery(queryName));
   }
 
-  public void configureIndices(List<FieldIndex> fieldIndices) {
+  @VisibleForTesting
+  Collection<FieldIndex> getFieldIndexes() {
+    return persistence.runTransaction("Get indexes", () -> indexManager.getFieldIndexes());
+  }
+
+  public void configureFieldIndexes(List<FieldIndex> newFieldIndexes) {
     persistence.runTransaction(
-        "Configure indices",
+        "Configure indexes",
         () -> {
-          // TODO(indexing): Disable no longer active indices
-          for (FieldIndex fieldIndex : fieldIndices) {
-            indexManager.addFieldIndex(fieldIndex);
+          List<FieldIndex> existingIndexes = new ArrayList<>(indexManager.getFieldIndexes());
+          Collections.sort(existingIndexes, FieldIndex.SEMANTIC_COMPARATOR);
+          Iterator<FieldIndex> existingIt = existingIndexes.iterator();
+          @Nullable FieldIndex existingValue = advanceIterator(existingIt);
+
+          List<FieldIndex> updatedIndexes = new ArrayList<>(newFieldIndexes);
+          Collections.sort(updatedIndexes, FieldIndex.SEMANTIC_COMPARATOR);
+          Iterator<FieldIndex> updatedIt = updatedIndexes.iterator();
+          @Nullable FieldIndex updatedValue = advanceIterator(updatedIt);
+
+          while (existingValue != null || updatedValue != null) {
+            boolean deleted = false;
+            boolean updated = false;
+
+            if (existingValue != null && updatedValue != null) {
+              int cmp = FieldIndex.SEMANTIC_COMPARATOR.compare(existingValue, updatedValue);
+              if (cmp < 0) {
```

Comment:
It is not obvious immediately why comparison result leads to `deleted` or `updated`. Can we add some comment? Ideally a high level comment for the method, and some implementation comment to explain details.

Reply:
I added comments to the more generic implementation, as well as tests.